### PR TITLE
3.3-beta-3 scoped query strings being emptied

### PR DESCRIPTION
**Pull request by htstudios**
Ref: https://github.com/cakephp/cakephp/issues/9210

### CODE REVIEW

**Comment by markstory** [comment]
This causes a [bunch of test failures](https://travis-ci.org/cakephp/cakephp/jobs/149907006).

File: src/View/Helper/PaginatorHelper.php [hunk 1]
```diff
@@ -486,7 +486,10 @@ public function generateUrl(array $options = [], $model = null, $full = false)
         ];
 
         if (!empty($this->_config['options']['url'])) {
-            $key = implode('.', array_filter(['options.url', Hash::get($paging, 'scope', null)]));
+            $key = 'options.url.?';
```
Comment (markstory) [inline src/View/Helper/PaginatorHelper.php]: Is the `?` supposed to be there?

Comment (htstudios) [inline src/View/Helper/PaginatorHelper.php]: Yes, it is part of the path down through the options nested array into the query string.
Without it won't fetch the query-string.

If I change the block to:
```php
        if (!empty($this->_config['options']['url'])) {
            $key = 'options.url.?';
            if (!empty($paging['scope'])) {
                $key .= '.' . $paging['scope'];
            }
            debug($this->_config);
            debug($key);
            debug($url);
            $url = array_merge($url, Hash::get($this->_config, $key, []));
        }
```

This is the output of the 3 debugs,

### Without Scoping

#### `$this->_config`

```php
[
	'options' => [
		'url' => [
			'?' => [
				'code' => 'da'
			]
		]
	],
	'templates' => [...]
]
```

#### `$key`

```php
'options.url.?'
```

### `$url`

```php
[
	'page' => (int) 1,
	'limit' => null,
	'sort' => 'Languages.label',
	'direction' => 'asc'
]
```

### With `['scope' => 'Languages']`

#### `$this->_config`

```php
[
	'options' => [
		'defaultModel' => 'Languages',
		'url' => [
			'?' => [
				'Languages' => [
					'code' => 'da',
					'sort' => 'label',
					'direction' => 'asc'
				]
			]
		]
	],
	'templates' => [...]
]
```

#### `$key`

```php
'options.url.?.Languages'
```

### `$url`

```php
[
	'page' => (int) 1,
	'limit' => null,
	'sort' => 'Languages.label',
	'direction' => 'asc'
]
```

I'll try to figure out why the tests do not pass.

**Comment by htstudios** [comment]
Closing because the issue seems to be with `Router::url()` and query string parameters.
Ref: https://github.com/cakephp/cakephp/issues/9210#issuecomment-237813455

**Pull request closed without merging**